### PR TITLE
I add some useful and powerful function (like search to select on listbox, and clear all selects menu, and fix some bug)

### PR DESCRIPTION
**Pull request by HHsomeHand**
1. fix 'menu -> view -> window Transparency' menu item cheked bugs

2. add menu item 'menu -> Edit -> Search listbox... ctrl + s' to search string from listbox [powerful]

3. add menu item 'menu -> Edit -> Clear All Selects' to clear all the selects

### CODE REVIEW

**Comment by HHsomeHand** [comment]
vs2022 auto convert " Spanish (Spain, International Sort) " to "西班牙语(西班牙，国际风格) " in the .rc files because I use Chinese lanuage setting in vs2022. I didn't change it manually and didn't found the problem when i pull requeset....

File: MessageCrackWiz.rc [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #undef APSTUDIO_READONLY_SYMBOLS
 
 /////////////////////////////////////////////////////////////////////////////
-// Spanish (Spain, International Sort) resources
+// ��������(�����������ʷ��) resources
```
Comment (hernandp) [inline MessageCrackWiz.rc]: Use international/English text, or add Chinese resources separately.

File: MessageCrackWiz.rc [hunk 2]
```diff
@@ -44,12 +44,12 @@ END
...
 #endif    // APSTUDIO_INVOKED
 
-#endif    // Spanish (Spain, International Sort) resources
+#endif    // ��������(�����������ʷ��) resources
 /////////////////////////////////////////////////////////////////////////////
 
 
 /////////////////////////////////////////////////////////////////////////////
-// Spanish (Argentina) resources
+// ��������(����͢) resources
```
Comment (hernandp) [inline MessageCrackWiz.rc]: Use international/English text, or add Chinese resources separately.

File: MessageCrackWiz.rc [hunk 3]
```diff
@@ -44,12 +44,12 @@ END
 
 #endif    // APSTUDIO_INVOKED
 
-#endif    // Spanish (Spain, International Sort) resources
+#endif    // ��������(�����������ʷ��) resources
```
Comment (hernandp) [inline MessageCrackWiz.rc]: Use international/English text, or add Chinese resources separately.

File: MessageCrackWiz.rc [hunk 4]
```diff
@@ -308,7 +312,7 @@ IDB_MDI_D               BITMAP                  ".\\bmp\\mdi_d.bmp"
 
 IDB_WINDOW_D            BITMAP                  ".\\bmp\\window_d.bmp"
 
-#endif    // Spanish (Argentina) resources
+#endif    // ��������(����͢) resources
```
Comment (hernandp) [inline MessageCrackWiz.rc]: Use international/English text, or add Chinese resources separately.

File: MessageCrackWiz.vcxproj [hunk 5]
```diff
@@ -13,17 +13,17 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{B0FB4944-A964-4C5C-96D6-72B568F2D938}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
-    <WindowsTargetPlatformVersion>10.0.17763.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (hernandp) [inline MessageCrackWiz.vcxproj]: Do not change default build tools.
New platforms must be added separately, or we must migrate to other build system such as CMake.

File: MsgCrackWizard.cpp [hunk 6]
```diff
@@ -818,6 +856,27 @@ BOOL CALLBACK FilterDlg_DlgProc(HWND hwndDlg, UINT msg, WPARAM wParam, LPARAM lP
...
+    int offset = -1;
+    if (g_fr.Flags & FR_DOWN)
+    {
+        offset = 1;
+    }
+
+    for (int i = beginPos; i < count && i >= 0; i += offset) {
+        ListBox_GetText(hListBox, i, strItemText);
+
+        // ����ʹ��wcsstr���������ַ���
```
Comment (hernandp) [inline MsgCrackWizard.cpp]: Use English/international text.

File: resource.h [hunk 7]
```diff
@@ -1,6 +1,6 @@
 //{{NO_DEPENDENCIES}}
-// Microsoft Visual C++ generated include file.
-// Used by MessageCrackWiz.rc
+// Microsoft Visual C++ ���ɵİ����ļ���
```
Comment (hernandp) [inline resource.h]: Same.

File: MessageCrackWiz.rc [hunk 8]
```diff
@@ -189,6 +189,9 @@ BEGIN
         MENUITEM "Copy &Function\tCtrl+F",      ID_EDIT_COPYFUNCTION
         MENUITEM SEPARATOR
         MENUITEM "Message Fi&lters...\tCtrl+L", ID_EDIT_MESSAGEFILTERS
+        MENUITEM SEPARATOR
+        MENUITEM "Search listbox...\tCtrl+S",   ID_EDIT_SEARCHLISTBOX
+        MENUITEM "Clear All Selects",           ID_EDIT_CLEARALLSELECTS
```
Comment (hernandp) [inline MessageCrackWiz.rc]: proper spelling would be 'Clear selection"  , use a shortcut key with the '&' prefix.

File: MsgCrackWizard.cpp [hunk 9]
```diff
@@ -818,6 +856,27 @@ BOOL CALLBACK FilterDlg_DlgProc(HWND hwndDlg, UINT msg, WPARAM wParam, LPARAM lP
 	}
 }
 
+int user_ListBox_FindString(HWND hListBox, int beginPos, const WCHAR* searchStr) {
+    int count = ListBox_GetCount(hListBox);
+    WCHAR strItemText[512] = { 0 };
```
Comment (hernandp) [inline MsgCrackWizard.cpp]: do not use magic numbers,  specify some constant like MAX_ITEM_TEXT

File: stdafx.h [hunk 10]
```diff
@@ -37,4 +37,5 @@ SOFTWARE.
 #include <string>
 #include <memory>
 #include <cassert>
+#include <commdlg.h>
```
Comment (hernandp) [inline stdafx.h:40]: Do we need commdlg.h?

**Review by hernandp — CHANGES_REQUESTED** [review]
Please address the comments and do the proper fixes.

File: stdafx.h (hunk 10 above)
Comment (HHsomeHand) [reply stdafx.h:40]: yes,  we need it to show FindText Common Dialog

**Comment by HHsomeHand** [comment]
In the latest commit, these issues are addressed

**Comment by hernandp** [comment]
thanks @HHsomeHand  I will check them later

**Comment by hernandp** [comment]
thanks for the modifications, I will merge them and review versioning in master. 

sorry for the late response, I was very busy.

**Pull request merged**